### ISSUE TITLE
Implement Gremlin based entities for Artifact, ArtifactCoordinates and ArtifactGroup

### ISSUE THREAD

**Issue by sbespalov** [strongbox/strongbox#1651]
# Task Description
Every enitty should have new reopository class with Gremlin based CRUD and Spring Data Neo4j based custom Cypher queryes. Repositories should be placed under `org.carlspring.strongbox.gremlin.repositories`. 

# Tasks

The following tasks will need to be carried out:
* [x] Rework `ArtifactEntry` class:
    - [x] Create `Artifact` interface
    - [x] Create `ArtifactEntity` class with `neo4j-ogm` annotations
    - [x] Implement `ArtifactAdapter`
    - [x] Rework `ArtifactEntryService` into `ArtifactRepository` 
    - [x] Replace all `ArtifactEntry` referencies with `ArtifactEntity`
* [x] Rework `AbstractArtifactCoordinates` class inheritors:
    - [x] rename `AbstractArtifactCoordinates` to `GenericArtifactCoordinates`
    - [x] Use `neo4j-ogm` annotations instead of `javax.persistence`
    - [x] Implement `ArtifactCoordinatesAdapter` and inheritors per concrete `ArtifactCoordinates` implementation
    - [x] Rework `ArtifactCoordinatesService` into  `ArtifactCoordinatesRepository`
* [x] Rework `ArtifactGroupEntry` and `RepositoryArtifactIdGroupEntry`
    - [x] Create `ArtifactGroup` and `RepositoryArtifactIdGroup` interfaces
    - [x] Create `ArtifactGroupEntity` class with `neo4j-ogm` annotations
    - [x] Create `RepositoryArtifactIdGroupEntity` class with `neo4j-ogm` annotations (see [RepositoryArtifactIdGroupEntity](https://github.com/strongbox/strongbox-janusgraph-cassandra-poc/blob/master/src/main/java/org/carlspring/strongbox/janusgraph/domain/RepositoryArtifactIdGroupEntity.java) as an inheritance example)
    - [x] Implement `ArtifactGroupAdapter`
    - [x] Implement `RepositoryArtifactIdGroupAdapter`
    - [x] Rework `ArtifactGroupService` into `ArtifactGroupRepository` 
    - [x] Rework `RepositoryArtifactIdGroupService` into `RepositoryArtifactIdGroupRepository` 
    - [x] Replace all `ArtifactGroupEntry` and `RepositoryArtifactIdGroupEntry` referencies with `ArtifactGroupEntity` and `RepositoryArtifactIdGroupEntity`
* [ ] Rework `org.carlspring.strongbox.data.criteria` API, it should be implemented with Gremlin

# Task Relationships

This task:
* Is a sub-task of: #1649
* depends on: #1650 

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @fuss86
  * @sbespalov
  * @steve-todorov

**Comment by sbespalov** [comment]
>  Rework org.carlspring.strongbox.data.criteria API, it should be implemented with Gremlin

this wil be done as followup task.

**Comment by carlspring** [comment]
Will be done as part of #1649 via #1676 .
